context_header.go

```diff
@@ -114,3 +114,19 @@ func WrapWithHeaders(ctx context.Context, headers map[string]string) ContextWith
 	newCtx := context.WithValue(ctx, contextKeyHeaders, h)
 	return headerCtx{Context: newCtx}
 }
+
+// WithoutHeaders hides any TChannel headers from the given context.
+func WithoutHeaders(ctx context.Context) context.Context {
+	return withoutHeaders{ctx}
```

Comment:
instead of another type, can we just use `context.WithValue` to override the current key to `nil`?